### PR TITLE
Use utf-8 for csv export

### PR DESCRIPTION
**Pull request by nebolax**
Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: rotkehlchen/tests/api/test_pnl_csv.py [hunk 1]
```diff
@@ -90,94 +92,104 @@ def assert_csv_export_response(response, csv_dir, is_download=False, unicode_che
...
+        return_value=encoding_to_use,
+    ):
+        # Add one event with unicode notes, to make sure unicode is exported in the CSV
+        with rotki.data.db.user_write() as write_cursor:
+            event_id = dbevents.add_history_event(
+                write_cursor=write_cursor,
+                event=EvmEvent(
+                    event_identifier=GENESIS_HASH,
+                    sequence_index=0,
+                    timestamp=1569924574 * 1000,
```
Comment (yabirgb) [inline rotkehlchen/tests/api/test_pnl_csv.py:119]: We have `ts_sec_to_ms` or you  can hardcode this one

File: rotkehlchen/tests/api/test_pnl_csv.py [hunk 2]
```diff
@@ -90,94 +92,104 @@ def assert_csv_export_response(response, csv_dir, is_download=False, unicode_che
...
+        # now query the export endpoint with json body
+        response = requests.get(
+            api_url_for(rotkehlchen_api_server_with_exchanges, 'historyexportingresource'),
+            json={'directory_path': csv_dir},
+        )
+        assert_csv_export_response(response, csv_dir, unicode_check=True)
+        # now query the export endpoint with query params
+        response = requests.get(
+            api_url_for(rotkehlchen_api_server_with_exchanges, 'historyexportingresource') +
+            f'?directory_path={csv_dir2}',
```
Comment (yabirgb) [inline rotkehlchen/tests/api/test_pnl_csv.py:148]: Since you are modifying this code can you update this to use function params instead of `.get` instead of concatenating the string? I believe the syntax is

```suggestion
            api_url_for(
                rotkehlchen_api_server_with_exchanges,
                'historyexportingresource',
                directory_path=csv_dir2,
            ) 
```

same in other places in the test

File: rotkehlchen/tests/api/test_pnl_csv.py [hunk 3]
```diff
@@ -90,94 +92,104 @@ def assert_csv_export_response(response, csv_dir, is_download=False, unicode_che
 @pytest.mark.parametrize('db_settings', [
     {'pnl_csv_with_formulas': False},
 ])
+@pytest.mark.parametrize('encoding_to_use', ['utf-8', 'cp1252'])
 def test_history_export_download_csv(
         rotkehlchen_api_server_with_exchanges,
         tmpdir_factory,
+        encoding_to_use,
 ):
```
Comment (yabirgb) [inline rotkehlchen/tests/api/test_pnl_csv.py:100]: This is a heavey tests due to the number of events processed and you only test the history export and (afaik) not the other places. Could you create a test only adding the event and trying to export it? In my pc this test takes around ~8 secs and you repeat it twice. I believe executing only the export that you need would be much faster. Also maybe you could also test the other locations where you export data if possible

File: docs/changelog.rst [hunk 4]
```diff
@@ -29,6 +29,7 @@ Changelog
 * :bug:`5749` Phishing zero token transfer transactions will now be hidden and ignored.
 * :bug:`5717` Swaps will now be processed in accounting correctly even if set manually.
 * :bug:`-` If a premium user changes their rotki password they will now be able to pull remote data without restarting the app.
+* :bug:`-` There will be no problem with exporting PnL reports that contain unusual (for example Japanese) characters.
```
Comment (yabirgb) [inline docs/changelog.rst:32]:
```suggestion
* :bug:`-` Fixed an issue where reports couldn't be exported if there was special characters in notes or assets and the user locale was not compatible.
```

**Review by yabirgb — COMMENTED** [review]
I believe we can improve testing speed by testing only the logic that you need

File: rotkehlchen/tests/api/test_pnl_csv.py [hunk 5]
```diff
@@ -177,7 +162,71 @@ def test_history_export_download_csv(
...
+        lambda start_ts, end_ts, has_premium: ('', [EvmEvent(
+            event_identifier=GENESIS_HASH,
+            sequence_index=0,
+            timestamp=TimestampMS(1569924574000),
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal(1)),
+            notes='Κοκκινολαίμης',  # Return an event with unicode characters
```
Comment (yabirgb) [inline rotkehlchen/tests/api/test_pnl_csv.py:194]: Sorry, I just recalled that lefteris mentioned that greek letters have a lower range in the unicode set and this could be a reason for the export to work with greek letters and not Chinese ones. Could you try adding also one of those to this note? https://www.khngai.com/chinese/charmap/tbluni.php?page=0

File: rotkehlchen/tests/api/test_pnl_csv.py [hunk 6]
```diff
@@ -160,14 +146,13 @@ def test_history_export_download_csv(
         tempzipfile.write_bytes(response.content)
         with zipfile.ZipFile(tempzipfile, 'r') as zip_ref:
             zip_ref.extractall(extractdir)
-        assert_csv_export_response(response, extractdir, is_download=True, unicode_check=True)
+        assert_csv_export_response(response, extractdir, is_download=True)
```
Comment (yabirgb) [inline rotkehlchen/tests/api/test_pnl_csv.py:149]: you can remove the logic for `unicode_check` also in `assert_csv_export_response`

**Review by yabirgb — COMMENTED** [review]
Thanks @nebolax ! Just some minor comments

File: rotkehlchen/tests/api/test_pnl_csv.py [hunk 7]
```diff
@@ -94,32 +92,16 @@ def test_history_export_download_csv(
...
-                event_type=HistoryEventType.SPEND,
-                event_subtype=HistoryEventSubType.FEE,
-                asset=A_ETH,
-                balance=Balance(amount=FVal(1)),
-                notes='Κοκκινολαίμης',
-            ))
-
+    """
+    Test that the csv export/download REST API endpoint works correctly.
+    Also tests that unicode is exported correctly into the CSV on different machines.
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_pnl_csv.py:97]: Is that correct? I don't see anything unicode specific here

```suggestion
```

**Review by LefterisJP — COMMENTED** [review]
Looks okay apart from the small comment. But tests failed:

```
FAILED rotkehlchen/tests/api/test_icons.py::test_upload_custom_icon[True-True-0-False] - FileNotFoundError: [Errno 2] No such file or directory: '/home/runner/work/rotki/rotki/frontend/app/public/assets/images/exchanges/kraken.svg'
FAILED rotkehlchen/tests/api/test_icons.py::test_upload_custom_icon[True-False-0-False] - AssertionError: Response contains unexpected status code. Details {'message': '{"file": ["Given path /home/runner/work/rotki/rotki/frontend/app/public/assets/images/exchanges/kraken.svg does not exist"]}', 'result': None}
FAILED rotkehlchen/tests/api/test_icons.py::test_upload_custom_icon_errors[True-True-0-False] - FileNotFoundError: [Errno 2] No such file or directory: '/home/runner/work/rotki/rotki/frontend/app/public/assets/images/exchanges/kraken.svg'
FAILED rotkehlchen/tests/api/test_icons.py::test_upload_custom_icon_errors[True-False-0-False] - FileNotFoundError: [Errno 2] No such file or directory: '/home/runner/work/rotki/rotki/frontend/app/public/assets/images/exchanges/kraken.svg'
FAILED rotkehlchen/tests/api/test_icons.py::test_refresh_icon[True] - FileNotFoundError: [Errno 2] No such file or directory: '/home/runner/work/rotki/rotki/frontend/app/public/assets/images/exchanges/kraken.svg'
```

I think they probably are fixed in develop so it just needs a rebase.

**Review by LefterisJP — APPROVED** [review]
LGTM
